sdk/iot/Azure.Iot.Hub.Service/src/Authentication/IotHubConnectionString.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Iot.Hub.Service.Authentication
+{
+    /// <summary>
+    /// Implementation for creating the shared access signature token provider.
+    /// </summary>
+    internal class IotHubConnectionString
+    {
+        public const string HostNameIdentifier = "HostName";
+        public const string SharedAccessKeyIdentifier = "SharedAccessKey";
+        public const string SharedAccessKeyNameIdentifier = "SharedAccessKeyName";
+        public const string SharedAccessSignatureIdentifier = "SharedAccessSignature";
+
+        private readonly ConnectionString _connectionString;
+        private readonly string _sharedAccessPolicy;
+        private readonly string _sharedAccessKey;
+        private readonly string _sharedAccessSignature;
+
+        public IotHubConnectionString(string connectionString)
+        {
+            _connectionString = ConnectionString.Parse(connectionString);
+            _sharedAccessKey = _connectionString.GetNonRequired(SharedAccessKeyIdentifier);
+            _sharedAccessPolicy = _connectionString.GetNonRequired(SharedAccessKeyNameIdentifier);
+            _sharedAccessSignature = _connectionString.GetNonRequired(SharedAccessSignatureIdentifier);
+
+            if (!ValidateInput(_sharedAccessPolicy, _sharedAccessKey, _sharedAccessSignature))
+            {
+                throw new ArgumentException("Specify either both the sharedAccessKey and sharedAccessKeyName, or only sharedAccessSignature");
+            }
+
+            HostName = _connectionString.GetRequired(HostNameIdentifier);
+        }
+
+        public string HostName { get; }
+
+        public ISasTokenProvider GetSasTokenProvider()
+        {
+            if (_sharedAccessSignature == null)
+            {
+                return new SasTokenProviderWithSharedAccessKey(HostName, _sharedAccessPolicy, _sharedAccessKey);
+            }
+            else
```

Comment:
you can get rid of the else and braces as the previous block returns.

Reply:
we will return either a SasTokenProviderWithSharedAccessKey or FixedSasTokenProvider, based on if the sharedAccessSignature is available in the connection string.